app/models/category.rb

```diff
@@ -26,4 +29,8 @@ def slug_candidates
   def normalize_friendly_id(input)
     input.to_s.to_slug.normalize.to_s
   end
+
+  def record_path
```

Comment:
don't put here , put it in banner model
although in banner model is not good, but we can refactor it later
banner model already use switch case(or if else)